### PR TITLE
[zh-cn] update the translation of "Handling common HTML and CSS problems"

### PR DESCRIPTION
**Pull request by hanyujie2002**
Relates to #19921

### CODE REVIEW

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md [hunk 1]
```diff
@@ -1,462 +1,281 @@
 ---
 title: 处理常见的 HTML 和 CSS 问题
 slug: Learn/Tools_and_testing/Cross_browser_testing/HTML_and_CSS
+l10n:
+  sourceCommit: a3d9f61a8990ba7b53bda9748d1f26a9e9810b18
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md:5]:
```suggestion
  sourceCommit: ec80e95431ed46f05946e9b12219fdd2962808e6
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md [hunk 2]
```diff
@@ -1,462 +1,281 @@
...
-一旦你确定了你将要使用的潜在问题技术列表，研究它们所支持的浏览器以及相关技术是否有用是一个不错的主意。请参阅下面的[查找帮助](/zh-CN/docs/preview-wiki-content#Finding_help)。
+一旦你确定了将要使用但不被普遍支持的技术列表，最好研究一下它们在哪些浏览器中受支持，以及相关的有用技巧。请参阅下面的[寻求帮助](#寻找帮助)。
 
 #### HTML 回退行为
 
-一些问题可以通过利用 HTML / CSS 的自然工作方式来解决。
+某些问题可以通过利用 HTML/CSS 的自然工作方式来解决。
 
-无法识别的 HTML 元素被浏览器视为匿名内联元素（有效内联元素，没有语义值，类似于 {{htmlelement("span")}} 元素）。你还可以参考他们通过自己的名字，并用 CSS 样式它们，例如 - 你只需要确保它们表现为你想让他们，例如对所有新的语义元素设置 `display: block;`（如 {{htmlelement("article")}}、{{htmlelement("aside")}} 等），但现在，基本只有在旧版本的 IE 不能识别他们（IE 8 和更低）。这样，新的浏览器就可以正常使用这些代码，但是旧的 IE 版本也可以对这些元素进行样式化。
+浏览器会将无法识别的 HTML 元素视为匿名内联元素（实际上是没有语义值的内联元素，类似于 {{htmlelement("span")}} 元素）。你仍然可以通过它们的名称引用它们，并使用 CSS 对其进行样式设置，例如——你只需要确保它们按照你希望的方式运行即可。你可以像处理其他元素一样对其进行样式设置，包括根据需要将 `display` 属性设置为非 `inline` 的值。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md:106]: #12632

```suggestion
浏览器会将无法识别的 HTML 元素视为匿名行级元素（实际上是没有语义值的行级元素，类似于 {{htmlelement("span")}} 元素）。你仍然可以通过它们的名称引用它们，并使用 CSS 对其进行样式设置，例如——你只需要确保它们按照你希望的方式运行即可。你可以像处理其他元素一样对其进行样式设置，包括根据需要将 `display` 属性设置为非 `inline` 的值。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md [hunk 3]
```diff
@@ -1,462 +1,281 @@
...
-
-```html
-<!--[if lte IE 8]>
-  <script src="ie-fix.js"></script>
-  <link href="ie-fix.css" rel="stylesheet" type="text/css" />
-<![endif]-->
-```
-
-只有当浏览器查看页面是 IE 8 或更旧时，此块才会应用 IE 特定的 CSS 和 JavaScript。lte 意思是“小于或等于”，但是也可以使用 lt、gt、gte、!(NOT) 以及其他逻辑语法。
+在这里，我们提供了一个 [RGB](/zh-CN/docs/Web/CSS/color_value/rgb){{cssxref("background-color")}}，它会在悬停时改变透明度，以提示用户按钮是交互式的，还提供了一些半透明的嵌入式 {{cssxref("box-shadow")}} 阴影，以赋予按钮一些质感和深度。尽管 RGB 颜色和盒子阴影目前已得到广泛支持，但它们并非一直存在——它们最早是在 IE9 中引入的。不支持 RGB 颜色的浏览器会忽略相关声明，这意味着在旧版浏览器中按钮背景就不会显示，文字也无法读取，这是非常不好的情况！
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md:188]:
```suggestion
在这里，我们提供了一个 [RGB](/zh-CN/docs/Web/CSS/color_value/rgb) {{cssxref("background-color")}}，它会在悬停时改变透明度，以提示用户按钮是交互式的，还提供了一些半透明的嵌入式 {{cssxref("box-shadow")}} 阴影，以赋予按钮一些质感和深度。尽管 RGB 颜色和盒子阴影目前已得到广泛支持，但它们并非一直存在——它们最早是在 IE9 中引入的。不支持 RGB 颜色的浏览器会忽略相关声明，这意味着在旧版浏览器中按钮背景就不会显示，文字也无法读取，这是非常不好的情况！
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md [hunk 4]
```diff
@@ -1,462 +1,281 @@
...
-nav,
-figure,
-figcaption {
-  display: block;
-}
-```
+为了解决这个问题，我们添加了第二个 `background-color` 声明，指定了一个十六进制颜色——这在非常老旧的浏览器中也得到支持，可以作为现代炫酷特性的后备方案。浏览器在访问此页面时，首先会应用第一个 `background-color` 值；当遇到第二个 `background-color` 声明时，如果浏览器支持 RGB 颜色，就会用这个值覆盖之前的值。如果不支持，它就会忽略整个声明，继续执行后续样式。
 
-然而，这并不是那么简单 - 你还需要通过 JavaScript 在 DOM 中创建你想要样式化的每个元素的副本，让它们变得可以风格化; 这是一个奇怪的怪癖，我们不会在这里让你知道细节。例如：
+> **备注：** 对于其他 CSS 特性，如 [media queries](/zh-CN/docs/Web/CSS/CSS_media_queries/Using_media_queries)、[`@font-face`](/zh-CN/docs/Web/CSS/@font-face) 和 [`@supports`](/zh-CN/docs/Web/CSS/@supports) 块来说，也是如此——如果不被支持，浏览器就会忽略它们。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md:194]:
```suggestion
> **备注：** 对于其他 CSS 特性，如[媒体查询](/zh-CN/docs/Web/CSS/CSS_media_queries/Using_media_queries)、[`@font-face`](/zh-CN/docs/Web/CSS/@font-face) 和 [`@supports`](/zh-CN/docs/Web/CSS/@supports) 块来说，也是如此——如果不被支持，浏览器就会忽略它们。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md [hunk 5]
```diff
@@ -1,462 +1,281 @@
...
-const asideElem = document.createElement('aside');
- ...
-```
+#### 选择器支持
 
-这听起来像是一个痛苦的处理，但幸运的是有一个 [polyfill](/zh-CN/docs/Glossary/polyfill) 可以为你做必要的修复，除此之外 - 请参阅 [HTML5Shiv](https://github.com/aFarkas/html5shiv) 了解所有细节（请参阅[手动安装](https://github.com/aFarkas/html5shiv#installation)以获取最简单的用法）。
+当然，如果你没有使用合适的[选择器](/zh-CN/docs/Learn/CSS/Building_blocks/Selectors)来选择你想要设置样式的元素，任何 CSS 特性都不会被应用！
 
-#### 选择器支持
+在一个以逗号分隔的选择器列表中，如果你只是错误地编写了一个选择器，它可能无法匹配任何元素。但是，如果有一个选择器是无效的，**整个**选择器列表以及整个样式块都将被忽略。因此，建议只在[可容错选择器列表](/zh-CN/docs/Web/CSS/Selector_list#可容错选择器列表）中使用带有 `:-moz-` 前缀的伪类或伪元素，例如 `:where(::-moz-thumb)`。请避免在逗号分隔的选择器组中使用带有 `:-moz-` 前缀的伪类或伪元素（除非它在 [`:is()`](/zh-CN/docs/Web/CSS/:is) 或 [`:where()`](/zh-CN/docs/Web/CSS/:where) 可容错选择器列表内），因为除了 Firefox 之外的所有浏览器都会忽略整个样式块。需要注意的是，`:is()` 和 `:where()` 都可以作为参数传递给其他选择器列表，包括 [`:has()`](/zh-CN/docs/Web/CSS/:has) 和 [`:not()`](/zh-CN/docs/Web/CSS/:not)。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md:200]:
```suggestion
在一个以逗号分隔的选择器列表中，如果你只是错误地编写了一个选择器，它可能无法匹配任何元素。但是，如果有一个选择器是无效的，**整个**选择器列表以及整个样式块都将被忽略。因此，建议只在[可容错选择器列表](/zh-CN/docs/Web/CSS/Selector_list#可容错选择器列表)中使用带有 `:-moz-` 前缀的伪类或伪元素，例如 `:where(::-moz-thumb)`。请避免在逗号分隔的选择器组中使用带有 `:-moz-` 前缀的伪类或伪元素（除非它在 [`:is()`](/zh-CN/docs/Web/CSS/:is) 或 [`:where()`](/zh-CN/docs/Web/CSS/:where) 可容错选择器列表内），因为除了 Firefox 之外的所有浏览器都会忽略整个样式块。需要注意的是，`:is()` 和 `:where()` 都可以作为参数传递给其他选择器列表，包括 [`:has()`](/zh-CN/docs/Web/CSS/:has) 和 [`:not()`](/zh-CN/docs/Web/CSS/:not)。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md [hunk 6]
```diff
@@ -1,462 +1,281 @@
...
 
-CSS 前缀带来的另外一个问题是 - 这些是一种用于允许浏览器供应商在技术处于试验状态时实现自己版本的 CSS（或 JavaScript）特性的机制，所以他们可以使用它来获取它没有与其他浏览器的实现冲突，或者最终的前置实现。举个例子：
+另一组问题来自 CSS 前缀——这种机制最初用于允许浏览器供应商在实验状态下实施自己版本的 CSS（或 JavaScript）特性，这样他们就可以试用并确保其正常工作，而不会与其他浏览器的实现或最终无前缀的实现发生冲突。
+
+例如，Firefox 使用 `-moz-`，而 Chrome/Edge/Opera/Safari 使用 `-webkit-`。你可能会在旧代码中遇到的其他（你可以安全删除）的前缀包括 `-ms-`（它由 Internet Explorer 和早期版本的 Edge 使用），以及 `-o`（它用于最初的 Opera 版本）。
 
-- Mozilla 使用 `-moz-`
-- Chrome / Opera / Safari 使用 `-webkit-`
-- 微软使用 `-ms-`
+前缀特性从未打算在生产网站中使用——它们可能会在没有警告的情况下发生变化或被移除，可能会导致需要这些特性的旧版本浏览器遇到性能问题，并且一直是跨浏览器问题的罪魁祸首。例如，当开发人员决定仅使用 `-webkit-` 版本的属性时，这意味着该网站将无法在其他浏览器上正常工作。这种情况实际上发生过无数次，以至于其他浏览器供应商为几种 CSS 属性实施了 `-webkit-` 前缀版本。虽然浏览器仍然支持某些前缀属性名称、属性值和伪类，但现在实验性功能被放在标志后面，以便网页开发人员在开发过程中进行测试。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md:222]:
```suggestion
请不要在用于生产的网站中使用前缀特性——它们可能会在没有警告的情况下发生变化或被移除，可能会导致需要这些特性的旧版本浏览器遇到性能问题，并且一直是跨浏览器问题的罪魁祸首。例如，当开发人员决定仅使用 `-webkit-` 版本的属性时，这意味着该网站将无法在其他浏览器上正常工作。这种情况实际上发生过无数次，以至于其他浏览器供应商为几种 CSS 属性实施了 `-webkit-` 前缀版本。虽然浏览器仍然支持某些前缀属性名称、属性值和伪类，但现在实验性特性需要使用标志来启用，以便网页开发人员在开发过程中进行测试。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md [hunk 7]
```diff
@@ -1,462 +1,281 @@
...
-5. 现在尝试为你感兴趣的 CSS 属性设置一个新的值，你可以使用元素的[style](/zh-CN/docs/Web/API/HTMLElement/style)属性来做到这一点，例如尝试在 JavaScript 控制台中输入这些：
+5. 现在尝试在该元素上设置新的 CSS 属性值；你可以使用元素的 [style](/zh-CN/docs/Web/API/HTMLElement/style) 属性来实现，例如试着在 JavaScript 控制台中输入：
 
    ```js
    test.style.transform = "rotate(90deg)";
-   test.style.webkitTransform = "rotate(90deg)";
    ```
 
-当你开始在第二个点之后键入属性名称表示形式（请注意，在 JavaScript 中，CSS 属性名称是以较低的驼峰大小写，而不是连字符），JavaScript 控制台应该开始自动填充浏览器中存在的属性的名称并匹配到目前为止所写的内容。这对于找出在该浏览器中实现哪个版本的属性很有用。
+当你开始在第二个点后输入属性名表示时（请注意，在 JavaScript 中，CSS 属性名是用{{Glossary("camel_case", "驼峰格式")}}而不是{{Glossary("kebab_case", "连字符格式")}})，JavaScript 控制台应该能够匹配你输入的内容，并自动完成目前浏览器中已存在的属性名。这可以帮助你发现该浏览器中实现了哪些属性。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/html_and_css/index.md:252]:
```suggestion
当你开始在第二个点后输入属性名表示时（请注意，在 JavaScript 中，CSS 属性名是用{{Glossary("camel_case", "小驼峰式命名法")}}而不是{{Glossary("kebab_case", "烤肉串式命名法")}}），JavaScript 控制台应该能够匹配你输入的内容，并自动完成目前浏览器中已存在的属性名。这可以帮助你发现该浏览器中实现了哪些属性。
```

**Review by yin1999 — COMMENTED** [review]
一些问题

**Review by yin1999 — APPROVED** [review]

**Pull request merged**